### PR TITLE
Elasticsearch TLS support

### PR DESCRIPTION
**Pull request by spmistry**
**What changed?**
Additional support for connection to Elastic search over TLS for v6 and v7 clients

**Why?**
Enhancements to support Elasticsearch providers enforcing TLS connections using signed certificates.

**How did you test it?**
Connecting to Elasticsearch deployment requiring TLS and pem files

**Potential risks**
None. If TLS config isn't set default behavior will be enforced.

**Release notes**
ES configuration will have an addition nested field for TLS config [here](https://github.com/uber/cadence/blob/12579bd153360679f8442a3589f35f279f19ec6b/common/config/tls.go#L25)

**Documentation Changes**
Added documentation in visibility-on-elasticsearch.md

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -118,7 +118,8 @@ This will run all the tests excluding end-to-end integration test in host/ packa
 ```bash
 make test
 ```
-> Note that you will see some test failures because of errors connecting to MySQL/Postgres if only Cassandra is up. This is okay if you don't write any code related to persistence layer. 
+:warning: Note:
```
Comment (longquanzheng) [inline CONTRIBUTING.md:121]: TIL! I didn't know that we can have a eye catching sign like this. Thank you!

File: common/config/elasticsearch.go [hunk 2]
```diff
@@ -48,6 +48,9 @@ type (
 		// optional to use AWS signing client
 		// See more info https://github.com/olivere/elastic/wiki/Using-with-AWS-Elasticsearch-Service
 		AWSSigning AWSSigning `yaml:"awsSigning"`
+		// optional to use Signed Certificates over https
+		// See more info https://github.com/olivere/elastic/wiki/Using-with-AWS-Elasticsearch-Service
```
Comment (longquanzheng) [inline common/config/elasticsearch.go:52]: This comment is not relevant.

File: docs/visibility-on-elasticsearch.md [hunk 3]
```diff
@@ -75,6 +75,25 @@ This part is used to config advanced visibility store to ElasticSearch.
  - `url` is for Cadence to discover ES 
  - `indices/visibility` is ElasticSearch index name for the deployment.  
 
+Optional TLS Support can be enabled by setting the TLS config as follows:
```
Comment (longquanzheng) [inline docs/visibility-on-elasticsearch.md:78]: Thanks for adding documentation. Can you also open a PR in cadence-docs to add the same docs in https://cadenceworkflow.io/docs/concepts/search-workflows/ ?

**Review by longquanzheng — APPROVED** [review]
LGTM, just some minor comments. 

Also since we don't have integ tests for this, can you try run it locally to make sure it works?

File: common/config/elasticsearch.go (hunk 2 above)
Comment (spmistry) [reply common/config/elasticsearch.go]: removed

File: docs/visibility-on-elasticsearch.md (hunk 3 above)
Comment (spmistry) [reply docs/visibility-on-elasticsearch.md:78]: Created: https://github.com/uber/cadence-docs/pull/40

**Review by longquanzheng — APPROVED** [review]

**Pull request merged**